src/main/drivers/adc_impl.h

```diff
@@ -28,20 +28,24 @@
 #define ADC_TAG_MAP_COUNT 10
 #endif
 
-typedef enum ADCDevice {
-    ADCINVALID = -1,
-    ADCDEV_1   = 0,
-#if defined(STM32F3) || defined(STM32F4) || defined(STM32F7)
-    ADCDEV_2,
-    ADCDEV_3
-#endif
-} ADCDevice;
-
 typedef struct adcTagMap_s {
     ioTag_t tag;
+#if !defined(STM32F1) // F1 pins have uniform connection to ADC instances
```

Comment:
Not really needed any more. ;-)

Reply:
I'm wondering if we can start explicitly dropping F1 support...

Reply:
I am wondering the same. The reality is I don't want to drop it, but if we keep doing extra work to not break F1 support, we don't gain anything from not publishing the F1 targets any more, and they will break at some stage anyway, and be it only for running out of flash space.

Reply:
I'm in favour of keeping the F1 build flags in the code, even though F1 is not supported.

F1 is not supported, but it is not withdrawn either. In other words we don't fix F1 specific bugs, or build problems, but if a user wants to build an F1 target themselves, we should not make that difficult/impossible.

Our workload is still reduced, since we don't need to fix F1 problems, in particular we don't have to keep doing the ROM reduction, which was one of the main time consumers.

Reply:
Agreed in principle with this course of action, @martinbudden. I think the crux of the matter is that keeping the F1 build flags in code when it gets changed, but not testing F1 specifically, will quickly lead to build problems for F1, and then break the F1 build.